route/src/route_state_worker.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "route_state_worker.h"
+
+namespace route {
+
+    RouteStateWorker::RouteStateWorker() : state_(RouteState::ROUTE_LOADING) { }
+
+    RouteStateWorker::RouteState RouteStateWorker::get_route_state() {
+        return state_;
+    }
+
+    void RouteStateWorker::on_route_event(RouteEvent event) {
+        switch (state_)
```

Comment:
It looks like the state machine itself is correct, but it's probably worth aligning the names of these states and events with the document. RouteEvent::LOAD_ROUTE_FILES vs route_loaded, for example. Either the document should be updated to match the code or vice versa.